docs/design/metadata.md

```diff
@@ -0,0 +1,365 @@
+   configuration. This list must be defined and non-empty.
+
+2. *Feature priorities* that override the order of features. The initial
+   order is determined by ordering the namespaces according to their
+   priorities (as indicated in point 1.), then replacing them by all
+   their features as ordered by the provider plugin. Afterwards, all
+   features defined in the feature priority list are moved to
+   the beginning of the resulting list, while the relative order
+   of the remaining items remains the same. Said order may be altered
+   by user configuration. This list is optional.
+
+3. *Property priorities* that override the order of properties.
+   The initial order is determined by ordering the features according
+   to their priorities (as indicated in point 2.), then replacing them
+   by all their properties as ordered by the provider plugin.
+   Afterwards, all properties defined in the feature priority list
+   are moved to the beginning of the resulting list, while the relative
+   order of the remaining items remains the same. Said order may be
+   altered by user configuration. This list is optional.
+
+### Provider information
+
+The wheel metadata also includes the provider metadata dictionary,
+keyed on namespaces. This dictionary is obligatory, and must include
+the metadata for all namespaces that are listed in default namespace
+priorities. The values include the following information:
+
+- The *plugin API* value that specifies how to load plugins. It is
+  obligatory. It follows the same format as object references
+  in the [entry points specification](
+  https://packaging.python.org/en/latest/specifications/entry-points/),
+  that is:
+
+  ```
+  importable.module:ClassName
+  ```
+
+  The plugin will be instantiated by the equivalent of:
+
+  ```python
+  import importable.module
+
+  plugin_instance = importable.module.ClassName()
+  ```
+
+- The list of *plugin requirements* that consists of zero or more
+  [dependency specifiers](
+  https://packaging.python.org/en/latest/specifications/dependency-specifiers/)
+  specifying how the provider should be installed. This is optional, but it is
+  necessary to support installing plugins in an isolated environment.
+
+- An optional *enable-if* clause that specifies an [environment marker](
+  https://packaging.python.org/en/latest/specifications/dependency-specifiers/#environment-markers)
+  specifying when the plugin should be used. If it specified, the plugin
+  will not be installed and loaded if the environment does not match
+  the specified markers. If it not specified, the plugin is enabled
+  unconditionally.
+
+
+## `pyproject.toml` file
+
+### Base information
+
+The `pyproject.toml` file is the standard project configuration file
+as defined in [pyproject.toml specification](
+https://packaging.python.org/en/latest/specifications/pyproject-toml/#pyproject-toml-spec).
+Variant wheels utilize a dedicated `variant` table.
```

Comment:
```suggestion
Variant-enabled packages utilize a dedicated top-level `variant` table.
```
(since `pyproject.toml` describes a package, and not build artifacts)